conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegration.java

```diff
@@ -230,95 +176,102 @@ public void reportPlaybackDeficiency(String message, Client.ErrorSeverity severi
      * Puts the session state in a notMonitored state.
      */
     public void pauseTracking() {
-        try {
-            client.adStart(sessionId, Client.AdStream.SEPARATE, Client.AdPlayer.SEPARATE, Client.AdPosition.PREROLL);
-            client.detachPlayer(sessionId);
-            Log.d(TAG, "Tracking paused.");
-        } catch (ConvivaException e) {
-            Log.e(TAG, e.getLocalizedMessage());
+        Log.d(TAG, "Will report ad break started: " + ConvivaSdkConstants.AdPlayer.SEPARATE + " " + ConvivaSdkConstants.AdType.CLIENT_SIDE);
+        convivaVideoAnalytics.reportAdBreakStarted(ConvivaSdkConstants.AdPlayer.SEPARATE, ConvivaSdkConstants.AdType.CLIENT_SIDE);
+        if(isSessionActive) {
+            convivaVideoAnalytics.reportPlaybackEnded();
+            isSessionActive = false;
         }
 
+        Log.d(TAG, "Tracking paused.");
     }
 
     /**
      * Puts the session state from a notMonitored state into the last one tracked.
      */
     public void resumeTracking() {
-        try {
-            client.attachPlayer(sessionId, playerStateManager);
-            client.adEnd(sessionId);
-            Log.d(TAG, "Tracking resumed.");
-        } catch (ConvivaException e) {
-            Log.e(TAG, e.getLocalizedMessage());
+        if(!isSessionActive) {
+            convivaVideoAnalytics.reportPlaybackRequested();
+            isSessionActive = true;
         }
+
+        convivaVideoAnalytics.reportAdBreakEnded();
+        Log.d(TAG, "Tracking resumed.");
+    }
+
+    /**
+     * This should be called when the app is resumed
+     */
+    public void reportAppForegrounded() {
+        Log.d(TAG, "appForegrounded");
+        ConvivaAnalytics.reportAppForegrounded();
     }
+
+    /**
+     * This should be called when the app is paused
+     */
+    public void reportAppBackgrounded() {
+        Log.d(TAG, "appBackgrounded");
+        if(isSessionActive) {
+            ConvivaAnalytics.reportAppBackgrounded();
+            isSessionActive = false;
```

Comment:
I'll investigate the background playback case a little more to answer this...

Reply:
thanks

Reply:
Thanks for looking into this. In general applications can perform playback even when backgrounded so ideally all playback controls(pause, resume, seek and events etc.) should be allowed in background mode. With the current logic of appBackGrounded and appForeGrounded logic, the playback session status is tied to app background/foreground status. IMO these are independent. `isSessionActive` seems to be used here to check the last application state (backgrounded or foregrounded) instead of the session state. Should we use a different variable like `isBackgrounded` instead?

Reply:
Yes, that makes sense. I'll check that in with the next changes.